CHANGES

```diff
@@ -1,6 +1,12 @@
 Changelog
 =========
 
+v0.7 (2025-04-11)
+-----------------
+
+Drop legacy way to hook and depend on standard beangulp hook functionality.
```

Comment:
I think we should be more explicit about the consequence for users, e.g.: 

```txt
Drop legacy way to hook and depend on standard beangulp hook functionality. 

This means users need to update their importers to use the beangulp library, as recommended for beancount v2 and as required for beancount v3. Old importers that still use functionality from beancount.ingest will no longer work.
```

Reply:
First off, I don't think we should make this a breaking change. Also, we need to be more specific in terminology here IMHO. AFAICT understanding what "standard beangulp hook" means requires reading the beangulp source code (since even the code comments refer to 2-tuples of (filename, entries) in places). So we should explicitly describe what a "hook" is.

Reply:
I'm wondering if we could also make a compromise, not keeping all the legacy functionality (e.g. drop apply_hooks), but still allowing to work with beancount 2 importers. e.g. by what you suggested with the wrap() method. I think people having to change there import.py is ok, having to adjust all importers to beangulp is where it is more painful.

Reply:
Just another point to add here, with the last release (v0.5 (2024-01-21)), the dependency is for beancount >= 3 and I haven't seen anyone raise an issue about this yet, so maybe just going forward is also ok.